script.js

```diff
@@ -1,4 +1,119 @@
+var AWAITING_USER_NAME = "awaiting user name";
+var PSP_GAME = "psp game";
+var currentGameMode = AWAITING_USER_NAME;
+var userName = "";
+
+console.log("global");
+
+var winCount = 0;
+var loseCount = 0;
+var drawCount = 0;
+
+var totalCount = function (winCount, loseCount, drawCount) {
+  return Number(winCount) + Number(loseCount) + Number(drawCount);
+};
+console.log("stats");
+
+var winPercentage = function () {
+  return Math.floor((Number(winCount) * 100) / totalCount);
+};
+
+console.log("percent");
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  if ((currentGameMode = AWAITING_USER_NAME)) {
+    userName = input;
+    currentGameMode = PSP_GAME;
+    myOutputValue =
+      "Welcome to the SPS club, " +
+      userName +
+      "!" +
+      "<br><br>" +
+      "Please start the game by choosing either scissors, paper or stone.";
+
+    console.log(main);
+  } else if ((currentGameMode = PSP_GAME)) {
+    var gamePlay = "";
+    var lowercaseInput = input.toLowerCase();
+    var gameNumber = generateRandomNumber();
+
+    if (gameNumber == 1) {
+      gamePlay = "scissors";
+    }
+    if (gameNumber == 2) {
+      gamePlay = "paper";
+    }
+    if (gameNumber == 3) {
+      gamePlay = "stone";
+    }
+    var myOutputValue =
+      "Tough luck," +
+      userName +
+      "<br> You lost! You played: " +
+      input +
+      ", the computer played: " +
+      gamePlay +
+      "<br> Try again! <br> Wins:" +
+      Number(winCount) +
+      " <br> Losses: " +
+      Number(loseCount) +
+      "Winninig percentage: " +
+      winPercentage;
+  }
+
+  if (
+    (gamePlay == "scissors" && lowercaseInput == "stone") ||
+    (gamePlay == "stone" && lowercaseInput == "paper") ||
+    (gamePlay == "paper" && lowercaseInput == "scissors")
+  ) {
+    myOutputValue =
+      "Wow!" +
+      userName +
+      " <br> You won! You played: " +
+      input +
+      ", the computer played: " +
+      gamePlay +
+      " <br> Well done! Please enter 'secret' to activate the new game mode! <br> Wins:" +
+      Number(winCount) +
+      " <br> Losses: " +
+      Number(loseCount) +
+      "Winninig percentage: " +
+      winPercentage;
```

Comment:
Likewise, you might realize the value stored in `winPercentage` isn't a number, but it prints out
```
function () {
  return Math.floor((Number(winCount) * 100) / totalCount);
};
```
instead. To execute a function (and then use its `return` value, which is this case is the `Math.floor((Number(winCount * 100 ...`, we have to write it as `winPercentage()` instead of `winPercentage`.

Just using the function name without the parentheses (eg. `winPercentage`) will print out its **definition** (which is the erroneous `function () { return Math.floor(...` output you are seeing now). When I talk about **executing a function** (eg. `winPercentage()`), I am referring to the fact that you are executing all of the lines within the function (as opposed to just printing out the contents of the function, like printing the lines in the function) and then **returning** the value (which should be a number) at the end.